### PR TITLE
Rejoin kani-verifer into the overall workspace

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

1. Rejoin `kani-verifier` into the overall workspace.
2. This exposed a bug where the PATH was being amended in compile test incorrectly and it's had `target/debug` in PATH before `scripts`. I solved this by borrowing "bookrunner's" approach of amending its own path once correctly, versus the existing compiletest approach of amending the path (erroneously, now) with each invocation.
3. I also added a mitigation to prevent that sort of bug causing dev environments to spam hundreds of http requests to github.

### Resolved issues:

towards #1045 

### Testing:

* How is this change tested? existing suite

* Is this a refactor change? sort of, but it exposed some bugs

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: src/kani-verifier/src/lib.rs [hunk 1]
```diff
@@ -70,6 +71,24 @@ fn appears_setup() -> bool {
...
+/// In dev environments, this proxy shouldn't be used.
+/// But accidentally using it (with the test suite) can fire off
+/// hundreds of HTTP requests trying to download a non-existent release bundle.
+/// So if we positively detect a dev environment, raise an error early.
+fn fail_if_in_dev_environment() -> Result<()> {
+    if let Ok(exe) = std::env::current_exe() {
+        if let Some(path) = exe.parent() {
+            if path.ends_with("target/debug") || path.ends_with("target/release") {
+                bail!(
+                    "Running the proxies from a development environment? These are just meant for releases."
```
Comment (zhassan-aws) [inline src/kani-verifier/src/lib.rs:83]: Can we clarify the error message a bit? Example: "Running a release-only executable, \<executable-name\>, from a development environment. Please remove executable from path and rerun."

**Review by zhassan-aws — COMMENTED** [review]
LGTM. The `releasebundle` check in CI failed though.

File: .dockerignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 .git
 firecracker
-target
+target/debug
```
Comment (celinval) [inline .dockerignore:3]: Why do we need to change this?

File: src/kani-verifier/src/lib.rs [hunk 3]
```diff
@@ -58,6 +58,7 @@ pub fn proxy(bin: &str) -> Result<()> {
             setup(None)
         }
     } else {
+        fail_if_in_dev_environment()?;
```
Comment (celinval) [inline src/kani-verifier/src/lib.rs:61]: How are we supposed to test this crate locally?

**Review by celinval — COMMENTED** [review]
Is it possible to use `cargo package` and `cargo install --path` in the CI to verify the installation behavior?

File: .dockerignore (hunk 2 above)
Comment (tedinski) [reply .dockerignore:3]: The Dockerfile is now copying `target/release/{kani,cargo-kani}` into the container.

File: src/kani-verifier/src/lib.rs (hunk 3 above)
Comment (tedinski) [reply src/kani-verifier/src/lib.rs:61]: We don't except by using the Dockerfile.

**Comment by tedinski** [comment]
> Is it possible to use `cargo package` and `cargo install --path` in the CI to verify the installation behavior?

Hmm, possibly. I will try this out and see.

File: .dockerignore (hunk 2 above)
Comment (tedinski) [reply .dockerignore:3]: Updated: Now `cargo install`ing the package, so we ignore `target/{release,debug}`, but need `target/package`

**Review by celinval — APPROVED** [review]
Nice. Thanks

**Pull request merged**